src/main/java/org/traccar/geocoder/OlaMapsGeocoder.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright 2012 - 2017 Anton Tananaev (anton@traccar.org)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.traccar.geocoder;
+
+import jakarta.json.JsonArray;
+import jakarta.json.JsonObject;
+import jakarta.json.JsonString;
+import jakarta.ws.rs.client.Client;
+
+public class OlaMapsGeocoder extends JsonGeocoder {
+
+    private static String formatUrl(String key) {
+        String url = "https://api.olamaps.io/places/v1/reverse-geocode?latlng=%f,%f";
+        if (key != null) {
+            url += "&api_key=" + key;
+        }
+        return url;
+    }
+
+    public OlaMapsGeocoder(Client client, String key, int cacheSize, AddressFormat addressFormat) {
+        super(client, formatUrl(key), cacheSize, addressFormat);
+    }
+
+    @Override
+    public Address parseAddress(JsonObject json) {
```

Comment:
It seems like format matched Google geocoder, so it doesn't make sense to duplicate it.

Reply:
Yes it's almost the same. How can we use this geocoder which has different url.

Reply:
You can probably modify Google geocoder to allow custom URL. But duplicating code is never a good solution.

Reply:
Ok will modify the PR.

Reply:
> You can probably modify Google geocoder to allow custom URL. But duplicating code is never a good solution.

I have done these changes. Kindly review.